File: tests/e2e/fixtures/code/sample.f
! Fortran: modules, derived types, allocatable arrays, do loops.
module scanner
  implicit none
  integer, parameter :: limit = 5 * 1024 * 1024

  type :: note
    character(len=64) :: path
    integer :: size = 0
  end type note
contains
  integer function count_large(notes) result(n)
    type(note), intent(in) :: notes(:)
    integer :: i
    n = 0
    do i = 1, size(notes)
      if (notes(i)%size > limit) n = n + 1
    end do
  end function count_large
end module scanner

program main
  use scanner
  type(note) :: notes(2) = [note('a.md', 12), note('b.md', 6291456)]
  print '(A,I0)', 'large: ', count_large(notes)
end program main

File: tests/e2e/fixtures/code/sample.pas
program Scanner;
{ Pascal: records, procedures, arrays, string handling. }
{$mode objfpc}{$H+}

uses SysUtils;

const
  Limit = 5 * 1024 * 1024;

type
  TNote = record
    Path: string;
    Size: Int64;
  end;
  TNotes = array of TNote;

function CountLarge(const Notes: TNotes): Integer;
var
  I: Integer;
begin
  Result := 0;
  for I := Low(Notes) to High(Notes) do
    if Notes[I].Size > Limit then
      Inc(Result);
end;

var
  Notes: TNotes;
begin
  SetLength(Notes, 2);
  Notes[0].Path := 'a.md'; Notes[0].Size := 12;
  Notes[1].Path := 'b.md'; Notes[1].Size := 6291456;
  WriteLn(Format('large: %d', [CountLarge(Notes)]));
end.
